apps/sms/js/selection_handler.js

```diff
@@ -89,15 +105,13 @@ SelectionHandler.prototype = {
   // and if all messages are checked : deselect them all.
   toggleCheckedAll: function sel_toggleCheckedAll() {
     var selected = this.selected.size;
-    var allInputs = this.getAllInputs();
-    var allSelected = (selected === allInputs.length);
+    var allInputs = this.inputIdSet;
+    var allSelected = (selected === allInputs.size);
```

Comment:
in the future, I'd really like a special case for "all selected" case. Instead of keeping all ids, I'd like a simple boolean. This would also help with case where the user taps "select all" while we're still loading the list.